### ISSUE TITLE
SDL3 cleanup: void functions

### ISSUE THREAD

**Issue by slouken** [libsdl-org/SDL#7233]
There are a number of functions that return void, but have failure conditions. SDL_SetWindowTitle(), SDL_SetWindowIcon(), etc.

For SDL3 we should change the return type to int and return -1 on error. The header documentation should be changed to:
```
 * \returns 0 on success or a negative error code on failure; call
 *          SDL_GetError() for more information.
```

**Comment by slouken** [comment]
Related, should we add an `SDL_ResultCode` to the API to make it easier for language bindings?

**Comment by 1bsyl** [comment]
```
SDL_assert.h:extern DECLSPEC void SDLCALL SDL_SetAssertionHandler(
SDL_assert.h:extern DECLSPEC void SDLCALL SDL_ResetAssertionReport(void);
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_AtomicLock(SDL_SpinLock *lock);
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_AtomicUnlock(SDL_SpinLock *lock);
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_MemoryBarrierReleaseFunction(void);
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_MemoryBarrierAcquireFunction(void);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_PlayAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_PauseAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_ClearAudioStream(SDL_AudioStream *stream);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_DestroyAudioStream(SDL_AudioStream *stream);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_MixAudioFormat(Uint8 * dst,
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_ClearQueuedAudio(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_LockAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_UnlockAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_CloseAudioDevice(SDL_AudioDeviceID dev);
SDL_error.h:extern DECLSPEC void SDLCALL SDL_ClearError(void);
SDL_events.h:extern DECLSPEC void SDLCALL SDL_PumpEvents(void);
SDL_events.h:extern DECLSPEC void SDLCALL SDL_FlushEvent(Uint32 type);
SDL_events.h:extern DECLSPEC void SDLCALL SDL_FlushEvents(Uint32 minType, Uint32 maxType);
SDL_events.h:extern DECLSPEC void SDLCALL SDL_SetEventFilter(SDL_EventFilter filter,
SDL_events.h:extern DECLSPEC void SDLCALL SDL_AddEventWatch(SDL_EventFilter filter,
SDL_events.h:extern DECLSPEC void SDLCALL SDL_DelEventWatch(SDL_EventFilter filter,
SDL_events.h:extern DECLSPEC void SDLCALL SDL_FilterEvents(SDL_EventFilter filter,
SDL_events.h:extern DECLSPEC void SDLCALL SDL_SetEventEnabled(Uint32 type, SDL_bool enabled);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_SetGamepadPlayerIndex(SDL_Gamepad *gamepad, int player_index);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_SetGamepadEventsEnabled(SDL_bool enabled);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_UpdateGamepads(void);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_CloseGamepad(SDL_Gamepad *gamepad);
SDL_guid.h:extern DECLSPEC void SDLCALL SDL_GUIDToString(SDL_GUID guid, char *pszGUID, int cbGUID);
SDL_haptic.h:extern DECLSPEC void SDLCALL SDL_HapticClose(SDL_Haptic * haptic);
SDL_haptic.h:extern DECLSPEC void SDLCALL SDL_HapticDestroyEffect(SDL_Haptic * haptic,
SDL_hidapi.h:extern DECLSPEC void SDLCALL SDL_hid_free_enumeration(SDL_hid_device_info *devs);
SDL_hidapi.h:extern DECLSPEC void SDLCALL SDL_hid_close(SDL_hid_device *dev);
SDL_hidapi.h:extern DECLSPEC void SDLCALL SDL_hid_ble_scan(SDL_bool active);
SDL_hints.h:extern DECLSPEC void SDLCALL SDL_ResetHints(void);
SDL_hints.h:extern DECLSPEC void SDLCALL SDL_DelHintCallback(const char *name,
SDL_hints.h:extern DECLSPEC void SDLCALL SDL_ClearHints(void);
SDL_init.h:extern DECLSPEC void SDLCALL SDL_QuitSubSystem(Uint32 flags);
SDL_init.h:extern DECLSPEC void SDLCALL SDL_Quit(void);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_LockJoysticks(void) SDL_ACQUIRE(SDL_joystick_lock);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_UnlockJoysticks(void) SDL_RELEASE(SDL_joystick_lock);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_SetJoystickPlayerIndex(SDL_Joystick *joystick, int player_index);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_GetJoystickGUIDString(SDL_JoystickGUID guid, char *pszGUID, int cbGUID);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_GetJoystickGUIDInfo(SDL_JoystickGUID guid, Uint16 *vendor, Uint16 *product, Uint16 *version, Uint16 *crc16);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_SetJoystickEventsEnabled(SDL_bool enabled);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_UpdateJoysticks(void);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_CloseJoystick(SDL_Joystick *joystick);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_ResetKeyboard(void);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_SetModState(SDL_Keymod modstate);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_StartTextInput(void);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_StopTextInput(void);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_ClearComposition(void);
SDL_keyboard.h:extern DECLSPEC void SDLCALL SDL_SetTextInputRect(const SDL_Rect *rect);
SDL_loadso.h:extern DECLSPEC void SDLCALL SDL_UnloadObject(void *handle);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogSetAllPriority(SDL_LogPriority priority);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogSetPriority(int category,
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogResetPriorities(void);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_Log(SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(1);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogVerbose(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogDebug(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogInfo(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogWarn(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogError(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogCritical(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogMessage(int category,
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogMessageV(int category,
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogGetOutputFunction(SDL_LogOutputFunction *callback, void **userdata);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogSetOutputFunction(SDL_LogOutputFunction callback, void *userdata);
SDL_main.h:extern DECLSPEC void SDLCALL SDL_SetMainReady(void);
SDL_main.h:extern DECLSPEC void SDLCALL SDL_UnregisterApp(void);
SDL_main.h:extern DECLSPEC void SDLCALL SDL_GDKSuspendComplete(void);
SDL_metal.h:extern DECLSPEC void SDLCALL SDL_Metal_DestroyView(SDL_MetalView view);
SDL_mouse.h:extern DECLSPEC void SDLCALL SDL_WarpMouseInWindow(SDL_Window * window,
SDL_mouse.h:extern DECLSPEC void SDLCALL SDL_SetCursor(SDL_Cursor * cursor);
SDL_mouse.h:extern DECLSPEC void SDLCALL SDL_DestroyCursor(SDL_Cursor * cursor);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroyMutex(SDL_mutex * mutex);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroySemaphore(SDL_sem *sem);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroyCond(SDL_cond *cond);
SDL_pixels.h:extern DECLSPEC void SDLCALL SDL_DestroyPixelFormat(SDL_PixelFormat *format);
SDL_pixels.h:extern DECLSPEC void SDLCALL SDL_DestroyPalette(SDL_Palette * palette);
SDL_pixels.h:extern DECLSPEC void SDLCALL SDL_GetRGB(Uint32 pixel,
SDL_pixels.h:extern DECLSPEC void SDLCALL SDL_GetRGBA(Uint32 pixel,
SDL_rect.h:extern DECLSPEC void SDLCALL SDL_GetRectUnion(const SDL_Rect * A,
SDL_rect.h:extern DECLSPEC void SDLCALL SDL_GetRectUnionFloat(const SDL_FRect * A,
SDL_render.h:extern DECLSPEC void SDLCALL SDL_UnlockTexture(SDL_Texture *texture);
SDL_render.h:extern DECLSPEC void SDLCALL SDL_RenderPresent(SDL_Renderer *renderer);
SDL_render.h:extern DECLSPEC void SDLCALL SDL_DestroyTexture(SDL_Texture *texture);
SDL_render.h:extern DECLSPEC void SDLCALL SDL_DestroyRenderer(SDL_Renderer *renderer);
SDL_rwops.h:extern DECLSPEC void SDLCALL SDL_DestroyRW(SDL_RWops * area);
SDL_sensor.h:extern DECLSPEC void SDLCALL SDL_CloseSensor(SDL_Sensor *sensor);
SDL_sensor.h:extern DECLSPEC void SDLCALL SDL_UpdateSensors(void);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_free(void *mem);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_GetOriginalMemoryFunctions(SDL_malloc_func *malloc_func,
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_GetMemoryFunctions(SDL_malloc_func *malloc_func,
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_aligned_free(void *mem);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_qsort(void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *));
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_DestroySurface(SDL_Surface *surface);
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_UnlockSurface(SDL_Surface *surface);
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_GetSurfaceClipRect(SDL_Surface *surface,
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_SetYUVConversionMode(SDL_YUV_CONVERSION_MODE mode);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_SetWindowsMessageHook(SDL_WindowsMessageHook callback, void *userdata);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_iPhoneSetEventPump(SDL_bool enabled);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_AndroidBackButton(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationWillTerminate(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationDidReceiveMemoryWarning(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationWillResignActive(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationDidEnterBackground(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationWillEnterForeground(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationDidBecomeActive(void);
SDL_system.h:extern DECLSPEC void SDLCALL SDL_OnApplicationDidChangeStatusBarOrientation(void);
SDL_thread.h:extern DECLSPEC void SDLCALL SDL_WaitThread(SDL_Thread * thread, int *status);
SDL_thread.h:extern DECLSPEC void SDLCALL SDL_DetachThread(SDL_Thread * thread);
SDL_thread.h:extern DECLSPEC void SDLCALL SDL_TLSCleanup(void);
SDL_timer.h:extern DECLSPEC void SDLCALL SDL_Delay(Uint32 ms);
SDL_timer.h:extern DECLSPEC void SDLCALL SDL_DelayNS(Uint64 ns);
SDL_version.h:extern DECLSPEC void SDLCALL SDL_GetVersion(SDL_version * ver);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowTitle(SDL_Window *window, const char *title);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowIcon(SDL_Window *window, SDL_Surface *icon);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowPosition(SDL_Window *window, int x, int y);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GetWindowPosition(SDL_Window *window, int *x, int *y);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowSize(SDL_Window *window, int w, int h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GetWindowSize(SDL_Window *window, int *w, int *h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GetWindowSizeInPixels(SDL_Window *window, int *w, int *h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowMinimumSize(SDL_Window *window, int min_w, int min_h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GetWindowMinimumSize(SDL_Window *window, int *w, int *h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowMaximumSize(SDL_Window *window, int max_w, int max_h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GetWindowMaximumSize(SDL_Window *window, int *w, int *h);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowBordered(SDL_Window *window, SDL_bool bordered);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowResizable(SDL_Window *window, SDL_bool resizable);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowAlwaysOnTop(SDL_Window *window, SDL_bool on_top);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_ShowWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_HideWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_RaiseWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_MaximizeWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_MinimizeWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_RestoreWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowGrab(SDL_Window *window, SDL_bool grabbed);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowKeyboardGrab(SDL_Window *window, SDL_bool grabbed);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_SetWindowMouseGrab(SDL_Window *window, SDL_bool grabbed);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_DestroyWindow(SDL_Window *window);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_EnableScreenSaver(void);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_DisableScreenSaver(void);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GL_UnloadLibrary(void);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GL_ResetAttributes(void);
SDL_video.h:extern DECLSPEC void SDLCALL SDL_EGL_SetEGLAttributeCallbacks(SDL_EGLAttribArrayCallback platformAttribCallback,
SDL_video.h:extern DECLSPEC void SDLCALL SDL_GL_DeleteContext(SDL_GLContext context);
SDL_vulkan.h:extern DECLSPEC void SDLCALL SDL_Vulkan_UnloadLibrary(void);

```

**Comment by 1bsyl** [comment]
That's the grep of all void returning function. maybe we could edit this previous messange and add sign  "+" before the one we should make return an error code.

I don't think we should add SDL_ResultCode, it's a boilerplate code.
But make a rule that that 0 on success, and < 0 on error. (not necessary -1)

**Comment by LiquidityC** [comment]
The following commands are a way of quickly skimming through possible offenders/candidates here:

- `grep -rn -A1 -P '(?<!return )SDL_InvalidParamError' src/* --color=always | less -R`
- `grep -rn -A1 -P '(?<!return )SDL_SetError' src/* --color=always | less -R`

If the "error set method" isn't followed by a return (all calls preceded by a return are already filtered) it might be a candidate. I can't find any more cases where `SDL_InvalidParamError` is called and the function doesn't return.

Still looking through the `SDL_SetError` cases. There are a few though so if anyone else is looking to get their fingers dirty you can probably dive in with little risk of merge conflicts.

**Comment by 1bsyl** [comment]
hey, I can also probably make a semantic patch that's spot all candidates and update function code and definition!
I'll give a try later

**Comment by 1bsyl** [comment]
see PR https://github.com/libsdl-org/SDL/pull/7255

**Comment by 1bsyl** [comment]
I've update the list (previous message). maybe we should turn more of them to return an error code.
like the one with pointer, or with ID

```
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_AtomicLock(SDL_SpinLock *lock);
SDL_atomic.h:extern DECLSPEC void SDLCALL SDL_AtomicUnlock(SDL_SpinLock *lock);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_ClearAudioStream(SDL_AudioStream *stream);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_DestroyAudioStream(SDL_AudioStream *stream);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_SetGamepadPlayerIndex(SDL_Gamepad *gamepad, int player_index);
SDL_gamepad.h:extern DECLSPEC void SDLCALL SDL_CloseGamepad(SDL_Gamepad *gamepad);
SDL_guid.h:extern DECLSPEC void SDLCALL SDL_GUIDToString(SDL_GUID guid, char *pszGUID, int cbGUID);
SDL_haptic.h:extern DECLSPEC void SDLCALL SDL_HapticClose(SDL_Haptic * haptic);
SDL_haptic.h:extern DECLSPEC void SDLCALL SDL_HapticDestroyEffect(SDL_Haptic * haptic,
SDL_hidapi.h:extern DECLSPEC void SDLCALL SDL_hid_free_enumeration(SDL_hid_device_info *devs);
SDL_hints.h:extern DECLSPEC void SDLCALL SDL_DelHintCallback(const char *name,
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_GetJoystickGUIDString(SDL_JoystickGUID guid, char *pszGUID, int cbGUID);
SDL_joystick.h:extern DECLSPEC void SDLCALL SDL_GetJoystickGUIDInfo(SDL_JoystickGUID guid, Uint16 *vendor, Uint16 *product, Uint16 *version, Uint16 *crc16);
SDL_loadso.h:extern DECLSPEC void SDLCALL SDL_UnloadObject(void *handle);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_Log(SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(1);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogVerbose(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogDebug(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogInfo(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogWarn(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogError(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogCritical(int category, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) SDL_PRINTF_VARARG_FUNC(2);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogGetOutputFunction(SDL_LogOutputFunction *callback, void **userdata);
SDL_log.h:extern DECLSPEC void SDLCALL SDL_LogSetOutputFunction(SDL_LogOutputFunction callback, void *userdata);
SDL_mouse.h:extern DECLSPEC void SDLCALL SDL_WarpMouseInWindow(SDL_Window * window,
SDL_mouse.h:extern DECLSPEC void SDLCALL SDL_DestroyCursor(SDL_Cursor * cursor);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroyMutex(SDL_mutex * mutex);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroySemaphore(SDL_sem *sem);
SDL_mutex.h:extern DECLSPEC void SDLCALL SDL_DestroyCond(SDL_cond *cond);
SDL_rwops.h:extern DECLSPEC void SDLCALL SDL_DestroyRW(SDL_RWops * area);
SDL_sensor.h:extern DECLSPEC void SDLCALL SDL_CloseSensor(SDL_Sensor *sensor);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_free(void *mem);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_GetOriginalMemoryFunctions(SDL_malloc_func *malloc_func,
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_GetMemoryFunctions(SDL_malloc_func *malloc_func,
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_aligned_free(void *mem);
SDL_stdinc.h:extern DECLSPEC void SDLCALL SDL_qsort(void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *));
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_DestroySurface(SDL_Surface *surface);
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_UnlockSurface(SDL_Surface *surface);
SDL_surface.h:extern DECLSPEC void SDLCALL SDL_GetSurfaceClipRect(SDL_Surface *surface,
SDL_system.h:extern DECLSPEC void SDLCALL SDL_SetWindowsMessageHook(SDL_WindowsMessageHook callback, void *userdata);
SDL_thread.h:extern DECLSPEC void SDLCALL SDL_WaitThread(SDL_Thread * thread, int *status);
SDL_thread.h:extern DECLSPEC void SDLCALL SDL_DetachThread(SDL_Thread * thread);
SDL_version.h:extern DECLSPEC void SDLCALL SDL_GetVersion(SDL_version * ver);
```

```
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_PlayAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_PauseAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_ClearQueuedAudio(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_LockAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_UnlockAudioDevice(SDL_AudioDeviceID dev);
SDL_audio.h:extern DECLSPEC void SDLCALL SDL_CloseAudioDevice(SDL_AudioDeviceID dev);

```

**Comment by sezero** [comment]
> I've update the list (previous message). maybe we should turn more of
> them to return an error code. like the one with pointer, or with ID

I really wouldn't touch `SDL_stdinc.h` procs, namely `SDL_qsort`, `SDL_free`
and `SDL_aligned_free`, because they their mimic standant libc counterparts.
As for `SDL_GetMemoryFunctions` and `SDL_GetOriginalMemoryFunctions`: what
is there to return?

**Comment by 1bsyl** [comment]
yes, this was just a 'grep'. removing std fucntions

**Comment by 1bsyl** [comment]
also log functions ?

**Comment by LiquidityC** [comment]
I can take a look through the list. I won't be fiddling with anything that looks volatile.
Some of the functions (gamepad eg.) can propagate errors after #7255 was merged so there are some low hanging fruits. I'll aim to close this issue with the next PR.

How does that sound?

**Comment by sezero** [comment]
> also log functions ?

AFAICS, yes.

**Comment by slouken** [comment]
> I can take a look through the list. I won't be fiddling with anything that looks volatile. Some of the functions (gamepad eg.) can propagate errors after #7255 was merged so there are some low hanging fruit. I'll aim to close this issue with the next PR.
> 
> How does that sound?

Sounds great, thanks!

**Comment by 1bsyl** [comment]
I can do the SDL_audio.h AudioID function

**Comment by 1bsyl** [comment]
here it is: https://github.com/libsdl-org/SDL/pull/7268/files

**Comment by LiquidityC** [comment]
Is there anything left to do here?

**Comment by madebr** [comment]
In https://github.com/libsdl-org/SDL/pull/7345, I dabbled a bit with `SDL_GetSystemRam`. On failure, it returns 0.
Should all functions returning integers be reviewed as well? Or does 0 make sense?

**Comment by LiquidityC** [comment]
> In #7345, I dabbled a bit with `SDL_GetSystemRam`. On failure, it returns 0. Should all functions returning integers be reviewed as well? Or does 0 make sense?

I hate when tickets/issues diverge and grow into other scopes. I'm sure the functions that return `int` could do with some looking over though. But Personally I'd prefer a new issue for the work.

For your particular function I think the documentation needs to be updated on the function. For Linux I guess `errno` will be set by `sysctl()` but that won'd be true for the other OS implementations (the VITA one is funny to me :smile:).

Looking at other cases I think we'd want to go for something like this for `SDL_GetSystemRAM()`:
```c
if (sysctl(mib, 2, &memsize, &len, NULL, 0) == 0) {             
    SDL_SystemRAM = (int)(memsize / (1024 * 1024));             
} else {                                                        
    return SDL_SetError("sysctl() failed: %s", strerror(errno));
}                                                               
```
**Eg.** We return -1 when it fails and >= 0 when nothing failed. Standard Linux behavior.

**Comment by slouken** [comment]
I think this is sorted out, and we decided not to add an SDL_ReturnCode. Thanks everyone!
